tahrir/endpoints/admin/badges.py

```diff
@@ -0,0 +1,56 @@
+from flask import abort, g, jsonify, request
+
+from ...app import csrf, oidc
+from ...utils.badge import convert_name_to_id
+from ...utils.user import require_admin
+from . import blueprint as bp
+
+
+@bp.route("/api/admin/badges", methods=["POST"])
+@csrf.exempt
+@oidc.require_login
+@require_admin
+def create_badge():
+    """Endpoint to create new badge"""
+
+    data = request.get_json()
+    if not data:
+        return abort(400, "No data provided")
+
+    required_fields = ["name", "image", "description", "criteria", "issuer_id"]
+    for field in required_fields:
+        if not data.get(field):
+            return abort(400, f"No data provided for {field}")
+
+    badge_id = convert_name_to_id(data.get("name"))
+
+    if not g.tahrirdb.badge_exists(badge_id):
+        g.tahrirdb.add_badge(
+            name=data.get("name"),
+            image=data.get("image"),
+            desc=data.get("description"),
+            criteria=data.get("criteria"),
+            issuer_id=data.get("issuer_id"),
+            tags=data.get("tags"),
+        )
+        return jsonify({"message": f"Badge {badge_id} created successfully"}), 201
+    else:
+        return abort(409, f"Badge with id {badge_id} already exists")
```

Comment:
This is minor, but in general I prefer to deal with the exception case early on so that the nominal case is always without indentation. Here it would be:
```py 
if g.tahrirdb.badge_exists(badge_id):
    return abort(409, f"Badge with id {badge_id} already exists")

g.tahrirdb.add_badge(
[...]
```

It makes it more readable because when a `else` statement happens far from its `if` statement it's harder to remember what the condition was.